jujuclient/utils.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jujuclient
+
+import (
+	"github.com/juju/errors"
+)
+
+func updateControllerModels(storedModels *ControllerModels, newModels map[string]ModelDetails) (bool, error) {
+	if len(storedModels.Models) == 0 && len(newModels) == 0 {
+		// Special case: no update necessary.
+		return false, nil
+	}
+	// Add or update controller models based on a new collection.
+	for modelName, details := range newModels {
+		if err := ValidateModel(modelName, details); err != nil {
+			// TODO what to do here?... don't want to stop
```

Comment:
Well, this validation will move outside of the update. So there would not be an error here :D